File: Iniciante/1040/1040.cs
using System; 

class URI {

    static void Main(string[] args) { 

string[] p = Console.ReadLine().Split(' ');
double n = double.Parse(p[0]);
double n1 = double.Parse(p[1]);
double n2 = double.Parse(p[2]);
double n3 = double.Parse(p[3]);
double medias = (n * 2.0 + n1 * 3.0 + n2 * 4.0 + n3 * 1.0) / 10.0;
if (medias == 4.85)
{
    medias = 4.8;
}
Console.WriteLine("Media: {0:f1}" ,medias);
bool fez_exame = false;
if (medias >= 5 && medias < 7)
{
    Console.WriteLine("Aluno em exame.");
    double exame = double.Parse(Console.ReadLine());
    Console.WriteLine("Nota do exame: {0:f1}" , exame);
    medias = (medias + exame) / 2;
    fez_exame = true;
}
if (medias >= 5)
{
    Console.WriteLine("Aluno aprovado.");
}
if (medias < 5)
{
    Console.WriteLine("Aluno reprovado.");
}
if (fez_exame)
{
    Console.WriteLine("Media final: {0:f1}" , medias);
}

    }

}

File: Iniciante/1040/1040.java
import java.io.IOException;
import java.util.Arrays;
import java.util.Scanner;
 
 
 
 
public class Main {
 
    public static void main(String[] args) throws IOException {
 
        Scanner nabana = new Scanner(System.in);
        double[] a = Arrays.stream(nabana.nextLine().split(" ")).mapToDouble(Double::parseDouble).toArray();
        float media = (float)(a[0]*2 + a[1]*3 + a[2]*4 + a[3])/10;
        System.out.println("Media: " + String.format("%.1f", media));

        if (media < 5) {
            System.out.println("Aluno reprovado.");
        }
        else if (media >= 5 && media < 7) {
            System.out.println("Aluno em exame.");
            String linha = nabana.nextLine();
            double c = Double.parseDouble(linha);
            System.out.println("Nota do exame: " + String.format("%.1f", c));
            double mediaFinal = (media + c) / 2;
            if (mediaFinal >= 5) {
                System.out.println("Aluno aprovado.");
            } else {
                System.out.println("Aluno reprovado.");
            }
            System.out.println("Media final: " + String.format("%.1f", mediaFinal));
        } else {
            System.out.println("Aluno aprovado.");
        }
        
        
        
        
        
 
    }
 
}